src/modules/projects/components/tables/ProjectHouseholdsTable.tsx

```diff
@@ -215,6 +181,36 @@ const ProjectHouseholdsTable = ({
       }}
       queryDocument={GetProjectHouseholdsDocument}
       columns={columns || defaultColumns}
+      TableBodyComponent={React.Fragment}
+      renderRow={(household, columnKeys) => {
+        return (
+          <TableBody
+            // Render each household as its own `tbody` in order to use the role='rowgroup' for better accessibility markup
+            key={household.id}
+            role='rowgroup'
+          >
+            <CustomDividerRow colSpan={(columns || defaultColumns).length} />
```

Comment:
should the colspan be determined by `columnKeys` since there may be optional columns?

Reply:
`columnKeys` is tricky because `key` is not required on the ColumnDef type, so that list includes only those columns which do provide keys. We could update GenericTable to so that `columnKeys` includes some representation of every column, either the key or the header, but that is also a bit tricky since the header could be a ReactNode.

We can't use `columnKeys` for the `belowRowsContent` below, unless we update that to accept a render function, which seems like too much cruft to me.

It seems ok to use `(columns || defaultColumns)` even with the optional column. It doesn't seem to do any harm for a `colSpan` to be provided that's longer than the width of the table.

Reply:
Ah okay. We should probably make `key` required if its used in this way. But that makes sense to me its OK if the colspan is longer. thanks!